x-pack/test/api_integration/deployment_agnostic/apis/observability/streams/lifecycle.ts

```diff
@@ -0,0 +1,293 @@
+
+      const updatedRootDefinition = await getStream(apiClient, 'logs');
+      expect((updatedRootDefinition as WiredReadStreamDefinition).stream.ingest.lifecycle).to.eql({
+        type: 'dlm',
+        data_retention: '999d',
+      });
+      expect(updatedRootDefinition.effective_lifecycle).to.eql({
+        type: 'dlm',
+        data_retention: '999d',
+        from: 'logs',
+      });
+    });
+
+    it('inherits dlm', async () => {
+      // create two branches, one that inherits from root and
+      // another one that overrides the root lifecycle
+      await putStream(apiClient, 'logs.inherits.lifecycle', wiredPutBody);
+      await putStream(apiClient, 'logs.overrides.lifecycle', wiredPutBody);
+
+      const rootDefinition = await getStream(apiClient, 'logs');
+      await putStream(apiClient, 'logs', {
+        dashboards: [],
+        stream: {
+          ingest: {
+            ...(rootDefinition as WiredStreamGetResponse).stream.ingest,
+            lifecycle: { type: 'dlm', data_retention: '10m' },
+          },
+        },
+      });
+      await putStream(apiClient, 'logs.overrides', {
+        dashboards: [],
+        stream: {
+          ingest: {
+            ...wiredPutBody.stream.ingest,
+            routing: [{ destination: 'logs.overrides.lifecycle', if: { never: {} } }],
+            lifecycle: { type: 'dlm', data_retention: '1d' },
+          },
+        },
+      });
+
+      await expectLifecycle(['logs.inherits', 'logs.inherits.lifecycle'], {
+        type: 'dlm',
+        data_retention: '10m',
+        from: 'logs',
+      });
+
+      await expectLifecycle(['logs.overrides', 'logs.overrides.lifecycle'], {
+        type: 'dlm',
+        data_retention: '1d',
+        from: 'logs.overrides',
+      });
+    });
+
+    it('applies the nearest parent lifecycle when deleted', async () => {
+      await putStream(apiClient, 'logs.10d', {
+        dashboards: [],
+        stream: {
+          ingest: {
+            ...wiredPutBody.stream.ingest,
+            lifecycle: { type: 'dlm', data_retention: '10d' },
+          },
+        },
+      });
+      await putStream(apiClient, 'logs.10d.20d', {
+        dashboards: [],
+        stream: {
+          ingest: {
+            ...wiredPutBody.stream.ingest,
+            lifecycle: { type: 'dlm', data_retention: '20d' },
+          },
+        },
+      });
+      await putStream(apiClient, 'logs.10d.20d.inherits', wiredPutBody);
+
+      // delete lifecycle of the 20d override
+      await putStream(apiClient, 'logs.10d.20d', {
+        dashboards: [],
+        stream: {
+          ingest: {
+            ...wiredPutBody.stream.ingest,
+            routing: [{ destination: 'logs.10d.20d.inherits', if: { never: {} } }],
+          },
+        },
+      });
+
+      await expectLifecycle(['logs.10d', 'logs.10d.20d', 'logs.10d.20d.inherits'], {
+        type: 'dlm',
+        data_retention: '10d',
+        from: 'logs.10d',
+      });
+    });
+
+    if (isServerless) {
```

Comment:
Isn't there some way to put these into a `describe` and adding tags instead to decide if that block should execute in hosted/serverless?

Reply:
I prefer the `if` over splitting them, makes it easier to find the tests.
@dmlemeshko Sorry for not checking the docs, but is there a way to tag tests to only run in serverless/hosted but keep it in the same file?

Reply:
Likewise I couldn't find any tag that facilitates this. +1 to keeping the `if` rather than not having the code co-located.

Reply:
More details on slack, but the recommended way is to split them which is not practical so we may see improvements to keep them in a single place in the future